client-mr/src/main/java/org/apache/hadoop/mapred/RssMapOutputCollector.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.hadoop.mapred;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.tencent.rss.common.util.ByteUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.io.RawComparator;
+import org.apache.hadoop.io.serializer.SerializationFactory;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RssMRConfig;
+import org.apache.hadoop.mapreduce.TaskCounter;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.api.records.ApplicationAttemptId;
+import org.apache.hadoop.yarn.api.records.ContainerId;
+
+import com.tencent.rss.client.api.ShuffleWriteClient;
+import com.tencent.rss.client.factory.ShuffleClientFactory;
+import com.tencent.rss.common.ShuffleServerInfo;
+import com.tencent.rss.common.exception.RssException;
+import com.tencent.rss.storage.util.StorageType;
+
+public class RssMapOutputCollector<K extends Object, V extends Object>
+    implements MapOutputCollector<K, V> {
+
+  private Task.TaskReporter reporter;
+  private Class<K> keyClass;
+  private Class<V> valClass;
+  private Set<Long> successBlockIds = Sets.newConcurrentHashSet();
+  private Set<Long> failedBlockIds = Sets.newConcurrentHashSet();
+  private int partitions;
+  private SortWriteBufferManager bufferManager;
+
+  @Override
+  public void init(Context context) throws IOException, ClassNotFoundException {
```

Comment:
too many code in init(), it should be separated in several methods.

Reply:
OK